### ISSUE TITLE
[BUG][Discover] "_source" Not Added to Side Nav on Removing All Columns with doc_table:hideTimeColumn Enabled

### ISSUE THREAD

**Issue by MadaniKK** [opensearch-project/OpenSearch-Dashboards#5538]
**Describe the bug**

**To Reproduce**
1. Go to `discover: advanced setting `
2. set` doc_table:hideTimeColum` to `ON`
3. Go to Discover
4. Select a column, then remove it in the canvas 
<img width="1214" alt="image" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/71857976/ff662203-fcc8-48a4-8041-7ccd594aa627">
<img width="1211" alt="image" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/71857976/09d21ed4-1667-40aa-af81-90502b5b8859">

**Expected behavior**
"_source" should be in the selected field when all columns are removed and no other field is selected.
**OpenSearch Version**
3.0.0

**Dashboards Version**
Please list the version of OpenSearch Dashboards being used.

**Plugins**

Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**
 - Browser and version [e.g. 22]

**Additional context**

**Comment by MadaniKK** [comment]
May I take this issue? Ty.

**Comment by ananzh** [comment]
In the legacy Discover, there is no time field. But in new Discover, we could see a time field in the left nav. 
<img width="1715" alt="Screenshot 2023-12-01 at 5 10 58 PM" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/79961084/f899b591-d00a-46d6-9532-27c28daa27d8">

But add it or remove it from left nav won't make any differences. The correct behavior should be, and pls help me verify
* No time field in side nav
* When display a time based index pattern, time field is displayed in the table by default
* When `doc_table:hideTimeColumn` is on, time field is not displayed in the table

**Comment by MadaniKK** [comment]
1. I suggest keeping the time field in the side nav bar because it exists in the legacy cover, and when the hide time column is off, we can still add the time column to the selected field and use drag-drop to arrange column order.
2. I think for the canvas, the `removeColumn` probably is off, it doesn't guarantee the column array with a safe fallback. I saw in the code that we added an extra buildColumn(columns) before rendering the canvas, which is why the canvas still behaves correctly when removing columns. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/3268e37223e762239176e0613f1c8b5f872000d5/src/plugins/discover/public/application/components/data_grid/data_grid_table.tsx#L59-L68
And this problem doesn't show when `hideTime` is off might be because the time field gets automatically added to columns every time we remove a column from canvas (just as described in issue #5527 ) I think adding a buildColumn(columns) before rendering the side bar like in the canvas code, would be an easy solution. But in the future, we should check with removeColumn functions and see why it didn't trigger the removing function with the safe fallback.
